eq-author/src/App/publish/PublishPage.test.js

```diff
@@ -0,0 +1,309 @@
+  });
+
+  it("should enable the 'Publish' button when all fields are populated", async () => {
+    const mocks = [
+      {
+        request: {
+          query: GET_PUBLISH_DETAILS,
+          variables: {
+            questionnaireId,
+          },
+        },
+        result: () => {
+          return {
+            data: {
+              questionnaire: {
+                id: questionnaireId,
+                publishDetails: {
+                  surveyId: "123",
+                  formType: { ONS: "456" },
+                  __typename: "Object",
+                },
+                __typename: "Object",
+              },
+            },
+          };
+        },
+      },
+    ];
+
+    const { getByTestId } = renderPublishPage(mocks);
+
+    await flushPromises();
+
+    expect(getByTestId("publish-survey-button").disabled).toBeFalsy();
+  });
+  it("should fire a query to publish the questionnaire when the 'Publish' button is pressed and clear the form", async () => {
+    const mocks = [
+      {
+        request: {
+          query: GET_PUBLISH_DETAILS,
+          variables: {
+            questionnaireId,
+          },
+        },
+        result: () => {
+          return {
+            data: {
+              questionnaire: {
+                id: questionnaireId,
+                publishDetails: {
+                  surveyId: "123",
+                  formType: { ONS: "456" },
+                  __typename: "Object",
+                },
+                __typename: "Object",
+              },
+            },
+          };
+        },
+      },
+      {
+        request: {
+          query: TRIGGER_PUBLISH_QUERY,
+          variables: {
+            input: {
+              questionnaireId: questionnaireId,
+              surveyId: "123",
+              formType: "456",
+            },
+          },
+        },
+        result: () => {
+          queryWasCalled = true;
+          return {
+            data: {
+              triggerPublish: {
+                id: "987-546-232",
+                launchUrl: "https://en.wikipedia.org/wiki/Spoon_theory",
+                __typename: "Object",
+              },
+            },
+          };
+        },
+      },
+      {
+        request: {
+          query: UPDATE_PUBLISH_DETAILS,
+          variables: {
+            input: {
+              questionnaireId: questionnaireId,
+              surveyId: "",
+              formType: { ONS: "" },
+            },
+          },
+        },
+        result: () => {
+          queryWasCalled = true;
+          return {
+            data: {
+              updatePublishDetails: {
+                id: questionnaireId,
+                publishDetails: {
+                  surveyId: "",
+                  formType: { ONS: "" },
+                  __typename: "Object",
+                },
+                __typename: "Object",
+              },
+            },
+          };
+        },
+      },
+    ];
+
+    window.alert = jest.fn();
```

Comment:
Do this in a before all block and make sure to set it back in an after all block otherwise it can cause issues in other tests.

```
let originalAlert;
beforeAll(() => {
	originalAlert = window.alert;
	window.alert = jest.fn();
});

afterAll(() => {
	window.alert = originalAlert;
});

```

Reply:
I was wondering about this as well, but doesnt Jest do anything to prevent tests from influencing each other?

Reply:
Technically all tests files are run in their own sandboxes so no this isn't necessary but if I try access this alert in a different test in this file it might not behave as expected and putting it at the top makes it easier to see whats going on.